caddy/directives.go

```diff
@@ -68,6 +68,23 @@ var directiveOrder = []directive{
 	{"browse", setup.Browse},
 }
 
+// RegisterDirective adds the given directive to caddy's list of directives.
+// Pass the name of a directive you want it to be placed after,
+// otherwise it will be placed at the bottom of the stack.
+func RegisterDirective(name string, setup SetupFunc, after string) {
+	dir := directive{name: name, setup: setup}
+	idx := len(directiveOrder)
+	for i := range directiveOrder {
+		if directiveOrder[i].name == after {
+			idx = i + 1
+			break
+		}
+	}
```

Comment:
I've been thinking about it a bit as well. What I'd really like to see is a numeric priority system so there is less "relative" ordering and more of a documented hierarchy, so nothing depends on any other directives existing. Putting at the top would probably be bad. You don't want things going above tls, root, and bind.